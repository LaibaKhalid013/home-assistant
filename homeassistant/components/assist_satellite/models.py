"""Models for assist satellite."""

from enum import IntFlag, StrEnum


class AssistSatelliteState(StrEnum):
    """Valid states of an Assist satellite entity."""

    LISTENING_WAKE_WORD = "listening_wake_word"
    """Device is streaming audio for wake word detection to Home Assistant."""

    LISTENING_COMMAND = "listening_command"
    """Device is streaming audio with the voice command to Home Assistant."""

    PROCESSING = "processing"
    """Home Assistant is processing the voice command."""

    RESPONDING = "responding"
    """Device is speaking the response."""


class AssistSatelliteEntityFeature(IntFlag):
    """Supported features of Assist satellite entity."""

    ANNOUNCE = 1
    """Device supports remotely triggered announcements."""
