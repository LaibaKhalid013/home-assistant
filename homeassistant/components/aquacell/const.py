"""Constants for the Aquacell integration."""

DOMAIN = "aquacell"
DATA_AQUACELL = "DATA_AQUACELL"
