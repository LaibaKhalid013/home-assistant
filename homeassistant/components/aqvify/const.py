"""Constants for the Aqvify integration."""

import logging

DOMAIN = "aqvify"

CONF_DEVICE_KEY = "device_key"

LOGGER = logging.getLogger(__package__)
