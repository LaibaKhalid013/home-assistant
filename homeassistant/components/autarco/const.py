"""Constants for the Autarco integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

DOMAIN: Final = "autarco"
LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(minutes=1)

CONF_PUBLIC_KEY: Final = "public_key"

SENSORS_SOLAR: Final = "solar"
