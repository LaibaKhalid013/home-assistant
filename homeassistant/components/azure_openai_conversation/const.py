"""Constants for the OpenAI Conversation integration."""

import logging

DOMAIN = "azure_openai_conversation"
LOGGER = logging.getLogger(__package__)

CONF_AZURE_OPENAI_RESOURCE = "azure_openai_resource"
CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "gpt-4"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 150
CONF_TOP_P = "top_p"
RECOMMENDED_TOP_P = 1.0
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
AZURE_OPEN_API_VERSION = "2024-02-01"
