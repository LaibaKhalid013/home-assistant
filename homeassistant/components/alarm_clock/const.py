"""Constants for the Alarm Clock integration."""

import voluptuous as vol

from homeassistant.const import CONF_ICON, CONF_NAME
import homeassistant.helpers.config_validation as cv

DOMAIN = "alarm_clock"
ENTITY_ID_FORMAT = DOMAIN + ".{}"

ATTR_ALARM_TIME = "alarm_time"
ATTR_REPEAT_DAYS = "repeat_days"
ATTR_NEXT_ALARM = "next_alarm"

CONF_ALARM_TIME = "alarm_time"
CONF_REPEAT_DAYS = "repeat_days"

EVENT_ALARM_CLOCK_STARTED = "alarm_clock.started"
EVENT_ALARM_CLOCK_FINISHED = "alarm_clock.finished"
EVENT_ALARM_CLOCK_CANCELLED = "alarm_clock.cancelled"
EVENT_ALARM_CLOCK_CHANGED = "alarm_clock.changed"

STORAGE_KEY = DOMAIN
STORAGE_VERSION = 1

STORAGE_FIELDS = {
    vol.Required(CONF_NAME): cv.string,
    vol.Optional(CONF_ICON): cv.icon,
    vol.Required(CONF_ALARM_TIME): cv.time,
    vol.Required(CONF_REPEAT_DAYS): cv.weekdays,
}
