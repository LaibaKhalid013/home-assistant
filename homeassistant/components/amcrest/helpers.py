"""Helpers for amcrest component."""

from __future__ import annotations

import logging
from typing import TYPE_CHECKING

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.helpers.typing import UndefinedType


def service_signal(service: str, *args: str) -> str:
    """Encode signal."""
    return "_".join([DOMAIN, service, *args])


def log_update_error(
    logger: logging.Logger,
    action: str,
    name: str | UndefinedType | None,
    entity_type: str,
    error: Exception,
    level: int = logging.ERROR,
) -> None:
    """Log an update error."""
    logger.log(
        level,
        "Could not %s %s %s due to error: %s",
        action,
        name,
        entity_type,
        error.__class__.__name__,
    )
