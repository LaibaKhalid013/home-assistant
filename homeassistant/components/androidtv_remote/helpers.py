"""Helper functions for Android TV Remote integration."""

from __future__ import annotations

from typing import TYPE_CHECKING

from androidtvremote2 import AndroidTVRemote

from homeassistant.helpers.storage import STORAGE_DIR

from .const import CONF_ENABLE_IME, CONF_ENABLE_IME_DEFAULT_VALUE

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant


def create_api(hass: HomeAssistant, host: str, enable_ime: bool) -> AndroidTVRemote:
    """Create an AndroidTVRemote instance."""
    return AndroidTVRemote(
        client_name="Home Assistant",
        certfile=hass.config.path(STORAGE_DIR, "androidtv_remote_cert.pem"),
        keyfile=hass.config.path(STORAGE_DIR, "androidtv_remote_key.pem"),
        host=host,
        loop=hass.loop,
        enable_ime=enable_ime,
    )


def get_enable_ime(entry: ConfigEntry) -> bool:
    """Get value of enable_ime option or its default value."""
    return entry.options.get(CONF_ENABLE_IME, CONF_ENABLE_IME_DEFAULT_VALUE)
