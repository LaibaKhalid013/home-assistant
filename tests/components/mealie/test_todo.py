"""Tests for the Mealie todo."""

from http import HTTPStatus
from unittest.mock import AsyncMock, patch

from aiomealie import MutateShoppingItem, ShoppingItem
import pytest
from requests.exceptions import HTTPError
from requests.models import Response
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration
from .conftest import make_shopping_item

from tests.common import MockConfigEntry, snapshot_platform


@pytest.fixture(autouse=True)
def platforms() -> list[Platform]:
    """Fixture to specify platforms to test."""
    return [Platform.TODO]


@pytest.fixture(name="api")
def mock_api(tasks: list[ShoppingItem]) -> AsyncMock:
    """Mock the api state."""
    api = AsyncMock()
    api.get_shopping_items.return_value = tasks
    return api


@pytest.fixture(name="mealie_api_status")
def mock_api_status() -> HTTPStatus | None:
    """Fixture to inject an http status error."""
    return None


@pytest.fixture(name="tasks")
def mock_tasks() -> list[MutateShoppingItem]:
    """Mock a mealie shopping item instance."""
    return [make_shopping_item()]


@pytest.fixture(autouse=True)
def mock_api_side_effect(
    api: AsyncMock, mealie_api_status: HTTPStatus | None
) -> MockConfigEntry:
    """Mock todoist configuration."""
    if mealie_api_status:
        response = Response()
        response.status_code = mealie_api_status
        api.get_tasks.side_effect = HTTPError(response=response)


async def test_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    entity_registry: er.EntityRegistry,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test todo entities."""
    with patch("homeassistant.components.mealie.PLATFORMS", [Platform.TODO]):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


async def test_get_todo_list_items(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test for getting a To-do list."""
    await setup_integration(hass, mock_config_entry)

    state = hass.states.get("todo.mealie_supermarket")
    assert state
    assert state.state == "3"


async def test_get_non_existant_todo_list_items(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test for getting a To-do Item."""
    await setup_integration(hass, mock_config_entry)

    mock_mealie_client.get_shopping_items.return_value = {
        frozenset(
            {
                "quantity": 1.0,
                "unit": None,
                "food": None,
                "note": "Soda",
                "isFood": False,
                "disableAmount": True,
                "display": "Soda",
                "shoppingListId": "27edbaab-2ec6-441f-8490-0283ea77585f",
                "checked": False,
                "position": 0,
                "foodId": None,
                "labelId": None,
                "unitId": None,
                "extras": {},
                "id": "f45430f7-3edf-45a9-a50f-73bb375090be",
                "label": None,
                "recipeReferences": [],
                "createdAt": "2024-06-25T10:45:03.362623",
                "updateAt": "2024-06-25T11:57:22.412650",
            }
        )
    }

    await hass.services.async_call(
        Platform.TODO,
        "get_items",
        {},
        target={"entity_id": "todo.mealie_supermarket"},
        blocking=True,
        return_response=True,
    )

    state = hass.states.get("todo.not_a_list")
    assert state is None


async def test_add_todo_list_item(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test for adding a To-do Item."""
    await setup_integration(hass, mock_config_entry)

    mock_mealie_client.add_shopping_item = AsyncMock()

    await hass.services.async_call(
        Platform.TODO,
        "add_item",
        {"item": "Soda"},
        target={"entity_id": "todo.mealie_supermarket"},
        blocking=True,
    )


async def test_update_todo_list_item(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test for adding a To-do Item."""
    await setup_integration(hass, mock_config_entry)

    mock_mealie_client.update_shopping_item = AsyncMock()

    await hass.services.async_call(
        Platform.TODO,
        "update_item",
        {"item": "aubergine", "rename": "Eggplant", "status": "completed"},
        target={"entity_id": "todo.mealie_supermarket"},
        blocking=True,
    )


async def test_delete_todo_list_item(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test for deleting a To-do Item."""
    await setup_integration(hass, mock_config_entry)

    mock_mealie_client.delete_shopping_item = AsyncMock()

    await hass.services.async_call(
        Platform.TODO,
        "remove_item",
        {"item": "aubergine"},
        target={"entity_id": "todo.mealie_supermarket"},
        blocking=True,
    )
