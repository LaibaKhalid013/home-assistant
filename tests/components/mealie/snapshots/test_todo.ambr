# serializer version: 1
# name: test_entities[todo.mealie_freezer-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'todo',
    'entity_category': None,
    'entity_id': 'todo.mealie_freezer',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:basket',
    'original_name': 'Freezer',
    'platform': 'mealie',
    'previous_unique_id': None,
    'supported_features': <TodoListEntityFeature: 15>,
    'translation_key': None,
    'unique_id': '01J0BC4QM2YBRP6H5G933CETT7_e9d78ff2-4b23-4b77-a3a8-464827100b46',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[todo.mealie_freezer-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mealie Freezer',
      'icon': 'mdi:basket',
      'supported_features': <TodoListEntityFeature: 15>,
    }),
    'context': <ANY>,
    'entity_id': 'todo.mealie_freezer',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '3',
  })
# ---
# name: test_entities[todo.mealie_special_groceries-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'todo',
    'entity_category': None,
    'entity_id': 'todo.mealie_special_groceries',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:basket',
    'original_name': 'Special groceries',
    'platform': 'mealie',
    'previous_unique_id': None,
    'supported_features': <TodoListEntityFeature: 15>,
    'translation_key': None,
    'unique_id': '01J0BC4QM2YBRP6H5G933CETT7_f8438635-8211-4be8-80d0-0aa42e37a5f2',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[todo.mealie_special_groceries-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mealie Special groceries',
      'icon': 'mdi:basket',
      'supported_features': <TodoListEntityFeature: 15>,
    }),
    'context': <ANY>,
    'entity_id': 'todo.mealie_special_groceries',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '3',
  })
# ---
# name: test_entities[todo.mealie_supermarket-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'todo',
    'entity_category': None,
    'entity_id': 'todo.mealie_supermarket',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:basket',
    'original_name': 'Supermarket',
    'platform': 'mealie',
    'previous_unique_id': None,
    'supported_features': <TodoListEntityFeature: 15>,
    'translation_key': None,
    'unique_id': '01J0BC4QM2YBRP6H5G933CETT7_27edbaab-2ec6-441f-8490-0283ea77585f',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[todo.mealie_supermarket-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mealie Supermarket',
      'icon': 'mdi:basket',
      'supported_features': <TodoListEntityFeature: 15>,
    }),
    'context': <ANY>,
    'entity_id': 'todo.mealie_supermarket',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '3',
  })
# ---
